pandas/_libs/src/klib/khash_python.h

```diff
@@ -13,25 +13,80 @@
 // is 64 bits the truncation causes collission issues.  Given all that, we use our own
 // simple hash, viewing the double bytes as an int64 and using khash's default
 // hash for 64 bit integers.
-// GH 13436
+// GH 13436 showed that _Py_HashDouble doesn't work well with khash
+// GH 28303 showed, that the simple xoring-version isn't good enough
+// thus murmur2-hash is used
+
+
+// specializations of https://github.com/aappleby/smhasher/blob/master/src/MurmurHash2.cpp
+// it is possible to have a special x64-version, which would need less operations, but 
+// using 32bit version always has also some benifits:
+//    - one code for 32bit and 64bit builds
+//    - the same case for 32bit and 64bit builds
+//    - no performance difference could be measured compared to a possible x64-version
+
+khint32_t PANDAS_INLINE murmur2_32_32to32(khint32_t k1, khint32_t k2){
```

Comment:
could we use this to implement int32/uint32/float32 hashtables?  (see #33287)

Reply:
@jbrockmendel I've a pet  project which wraps `khash` and for 32bit there is another specialization: https://github.com/realead/cykhash/blob/master/src/cykhash/murmurhash.pxi#L14 which has 6 operations less than  `murmur2_32_32to32`.

Reply:
Not gonna argue with fewer operations, but the big wins available if we can get #33287 are a) avoiding a _ton_ of cast/copy operations b) some code simplifications (in particular in index.pyx)

I don't want to hijack this thread, so if you're up for it, let's discuss over in #33287.